terraform/linter/linters.go

```diff
@@ -0,0 +1,263 @@
+package linter
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform/config"
+	"github.com/pkg/errors"
+)
+
+const (
+	validNameRegex = `^[a-z0-9_]+$`
```

Comment:
Can we extend it to not allow `_` as a prefix or suffix, and only allow a single `_` at a time.

Example: https://regex101.com/r/BeBFQx/1